xerces-c 3.2.0

rleigh-codelibre opened the pull request:
- [Y] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [Y] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [Y] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [Y] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Note, this release switches to cmake, and this automatically enables the use of `char16_t` for unicode strings in place of `uint16_t`.  This means you can use `u"string"` directly which is a nice usability gain, but it does mean that all packages using xerces-c must compile with C++11 enabled.  If this proves to be a problem, we can explicitly disable it until Xcode enables it by default.  I've patched xalan-c (https://issues.apache.org/jira/browse/XALANC-773).

ilovezfs commented on Formula/xalan-c.rb:
```diff
@@ -25,6 +26,7 @@ class XalanC < Formula
 
   def install
     ENV.deparallelize # See https://issues.apache.org/jira/browse/XALANC-696
+    ENV["CXXFLAGS"] = "-std=c++11"
```
You'll want to use

```
  needs :cxx11
```
and
```
    ENV.cxx11
```
instead.

ilovezfs commented on Formula/xalan-c.rb:
```diff
@@ -20,11 +21,16 @@ class XalanC < Formula
   def install
+    ENV.cxx11
     ENV.deparallelize # See https://issues.apache.org/jira/browse/XALANC-696
+    ENV["CXXFLAGS"] = "-std=c++11"
```
this shouldn't be needed now that you've added ENV.cxx11.

rleigh-codelibre replied:
D'oh.  I'll push that in just a second.

rleigh-codelibre commented:
@ilovezfs I've updated the revision of the dependent packages and added C++11 support there, but it's not building them in the CI testing AFAICT.  Is something else missing?  The errors when testing the bottled version are because the soname changed.

ilovezfs commented:
@rleigh-codelibre the ones that are failing need to be revision bumped.

ilovezfs commented:
@rleigh-codelibre I have pushed the revision bumps.

rleigh-codelibre commented:
@ilovezfs Thanks.  Some of them might need C++11 enabling.  And some like zquilla need some patches to fix incompatibilities in the code.  I'm working my way through these in the BSD ports tree, and I'll carry the patches over to homebrew once I've tested them.

rleigh-codelibre commented:
@ilovezfs I've patched all the necessary packages and filed bug reports upstream wherever possible.  The remaining errors are not in this PR but are warnings about existing problems:

```
11:25:34 ==> FAILED
11:25:34 Error: 1 problem in 1 formula
11:25:34 xml-tooling-c:
11:25:34   * Dependency curl should not use option with-openssl

11:54:55 ==> FAILED
11:54:55 Error: 1 problem in 1 formula
11:54:55 enigma:
11:54:55   * Dependency sdl_mixer should not use option with-libmikmod


12:06:16 ==> FAILED
12:06:16 Error: 1 problem in 1 formula
12:06:16 shibboleth-sp:
12:06:16   * Dependency curl should not use option with-openssl
```

Should these problems be addressed in this PR?  I'm not exactly sure what the fixes are for these audit violations.

ilovezfs commented:
>Should these problems be addressed in this PR? I'm not exactly sure what the fixes are for these audit violations.

Nope. They're being tracked in https://github.com/Homebrew/homebrew-core/issues/13133.

>I've patched all the necessary packages and filed bug reports upstream wherever possible. 

Please submit PR(s) to https://github.com/Homebrew/formula-patches so they don't have to live in the formulae themselves.

rleigh-codelibre commented:
@ilovezfs I've opened https://github.com/Homebrew/formula-patches/pull/175 .  I'll update this PR once that one is merged?

JCount commented:
@rleigh-codelibre I've merged all the patches, and I apologize for pinging you repeatedly whilst doing so. However, you should be set to move this PR forward. Thank you for your patience.

ilovezfs commented:
@rleigh-codelibre ping. Also note that this PR should be 1 commit per formula.

rleigh-codelibre commented:
Thanks.  I'm away for the next week, so I'll update this in 8 days.

ilovezfs commented:
@BrewTestBot test this please

ilovezfs commented:
@BrewTestBot test this please

dunn commented:
Looks like this needs to be rebased.

ilovezfs commented:
Right. But rleigh-codelibre is away for a few days, so …

rleigh-codelibre commented:
@JCount @ilovezfs Sorry for the delay; I've updated the branch to the the formula-patches and split up the commits.  Hope that's all OK.

rleigh-codelibre commented:
There's one audit failure which seems to be in shibboleth-sp related to curl/openssl, unrelated to this PR.

ilovezfs commented:
@BrewTestBot test this please

ilovezfs commented:
@gvenzl do you have any idea who we should report the patch for dbxml to? (see https://raw.githubusercontent.com/Homebrew/formula-patches/master/dbxml/c%2B%2B11.patch)

ilovezfs commented:
@BrewTestBot test this please

ilovezfs commented:
@BrewTestBot test this please

ilovezfs commented:
Thanks @rleigh-codelibre! You're awesome :)

:sparkle: :tada: :confetti_ball:
